components/Tabs/Tabs.js

```diff
@@ -1,63 +1,76 @@
 class TabLink {
-  constructor(tabElement){
+  constructor(tabElement) {
     // assign this.tabElement to the tabElement DOM reference
-    // this.tabElement;
-    
+    this.tabElement = tabElement;
+
     // Get the `data-tab` value from this.tabElement and store it here
-    // this.tabData = ; 
-    
-    // We need to find out if a user clicked 'all' cards or a specific category.  Follow the instructions below to accomplish this task:    
-    
-    /* <- Delete this comment block when you work on the if statement
-    // Check to see if this.tabData is equal to 'all'
-    if(){
+    this.tabData = this.tabElement.dataset.tab;
+
+    // We need to find out if a user clicked 'all' cards or a specific category.  Follow the instructions below to accomplish this task:
+
+    // <- Delete this comment block when you work on the if statement
+    // Check to see if this.tabData is equal to 'all' NOTE: have to set it equal to the string all as per line 11. (clicked 'all' )
+    // https://javascript.info/ifelse
+    if (this.tabData === "all") {
       // If `all` is true, select all cards regardless of their data attribute values
-      // this.cards = ;
+      this.cards = document.querySelectorAll(".card");
     } else {
       // else if `all` is false, only select the cards with matching this.tabData values
-      // this.cards = ;
+      this.cards = document.querySelectorAll(
+        `.card[data-tab='${this.tabData}']`
+      );
     }
-    /* <- Delete this comment block when you work on the if statement
+    // Delete this comment block when you work on the if statement
 
-     // Map over the newly converted NodeList we just created in our if statement above. Convert each this.cards element into a new instance of the TabCard class. Pass in a card object to the TabCard class. 
-    // this.cards = Array.from(this.cards).map();
+    // Map over the newly converted NodeList we just created in our if statement above. Convert each this.cards element into a new instance of the TabCard class. Pass in a card object to the TabCard class.
+    this.cards = Array.from(this.cards).map(function(tabcard) {
+      return new TabCard(tabcard);
+    });
 
     // Add a click event that invokes this.selectTab
-    // this.tabElement.addEventListener();
+    this.tabElement.addEventListener("click", this.selectTab.bind(this));
```

Comment:
You'll definitely see this `in the wild`, I prefer to use arrow syntax so you don't need to worry about .bind
```
this.tabElement.addEventListener("click", () => this.selectTab());
```